### PR TITLE
add typescript support

### PR DESCRIPTION
**Pull request by niklabh**
#### Summary
- Add typescript support for writing files with ts and tsx extensions

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/12019

https://mattermost.atlassian.net/browse/MM-18155

### CODE REVIEW

File: webapp/.eslintrc.json [hunk 1]
```diff
@@ -632,5 +632,47 @@
         "onlyEquality": false
       }
     ]
-  }
+  },
+  "overrides": [
+    {
+      "files": ["**/*.tsx", "**/*.ts"],
+      "extends": [
```
Comment (jespino) [inline webapp/.eslintrc.json]: This is not working properly, but I don't know why is not failing in mattermost-webapp. I'm investigating it.

Comment (jespino) [reply webapp/.eslintrc.json]: My fault I hadn't upgrade to eslint 6 in my repo.

File: webapp/.eslintrc.json [hunk 2]
```diff
@@ -632,5 +632,47 @@
...
+        "./node_modules/eslint-config-mattermost/.eslintrc.json",
```
Comment (jespino) [inline webapp/.eslintrc.json]: I think this shouldn't be here. We are extending the previous existing eslint config, so here we only should add the `"plugin:@typescript-eslint/recommended"` line

File: webapp/.eslintrc.json (hunk 1 above)
Comment (jespino) [inline webapp/.eslintrc.json]: Here we can replace it directly with the typescript plugin setting.
```suggestion
      "extends": "plugin:@typescript-eslint/recommended",
```

File: webapp/.eslintrc.json (hunk 2 above)
Comment (niklabh) [reply webapp/.eslintrc.json]: removed

File: webapp/.eslintrc.json (hunk 1 above)
Comment (niklabh) [reply webapp/.eslintrc.json]: done

**Pull request merged**